infra/pySourceCredChart/setup.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import print_function
+from setuptools import setup, find_packages, Command
+from setuptools.command.sdist import sdist
+from setuptools.command.build_py import build_py
+from setuptools.command.egg_info import egg_info
+from subprocess import check_call
+import os
+import sys
+import platform
+
+here = os.path.dirname(os.path.abspath(__file__))
+node_root = os.path.join(here, 'js')
+is_repo = os.path.exists(os.path.join(here, '.git'))
+
+npm_path = os.pathsep.join([
+    os.path.join(node_root, 'node_modules', '.bin'),
+                os.environ.get('PATH', os.defpath),
+])
+
+from distutils import log
+log.set_verbosity(log.DEBUG)
+log.info('setup.py entered')
+log.info('$PATH=%s' % os.environ['PATH'])
+
+LONG_DESCRIPTION = 'A python library for Source Cred Charts'
+
+def js_prerelease(command, strict=False):
```

Comment:
was this code copy/pasted from somewhere? if so can we (ideally) import it (if not) attribute it?

Reply:
cookiecutter was used for creating the ipywidget. I'll put that in the readme file and this particular file. 

The setup file is for anyone that is developing the python library. It needs to go through this setup.py to make sure all the dependencies are loaded and the py/JS communication is valid.  The file is not necessary to build the library, but not to install or use it. So, we don't need to import it, a developer just needs to run it when they want to rebuild the library.